### PR TITLE
[FX] Normalize torch. arithmetic operators to python operator.*

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54236 [FX] Normalize torch. arithmetic operators to python operator.***

Add a pass to normalize operator targets. Currently it resolves `operator.*` ops to the correspoinding `torch` ops when we can statically reason that the operator would dispatch to a torch op

Differential Revision: [D27149411](https://our.internmc.facebook.com/intern/diff/D27149411)

### CODE REVIEW

File: torch/fx/experimental/normalize.py [hunk 1]
```diff
@@ -117,3 +120,52 @@ def _args_kwargs_to_normalized_kwargs(self, sig : inspect.Signature, args : Tupl
...
+
+        traced = NormalizeOperators(traced).transform()
+    """
+    binary_magic_method_remap : Dict[Callable[[Any, Any], Any], Callable[[Any, Any], Any]] = {
+        torch.add : operator.add,
+        torch.mul : operator.mul,
+        torch.sub : operator.sub,
+        torch.div : operator.truediv,
+        torch.floor_divide : operator.floordiv,
+        torch.remainder : operator.mod,
```
Comment (zdevito) [inline torch/fx/experimental/normalize.py:148]: double check % does the same thing, since they are not exactly the same operator (requires testing with negative numbers) (https://rob.conery.io/2018/08/21/mod-and-remainder-are-not-the-same/)

File: torch/fx/experimental/schema_type_annotation.py [hunk 2]
```diff
@@ -33,6 +33,24 @@ def __init__(self, module : torch.nn.Module, annotate_functionals : bool = True,
...
         self.annotate_get_attrs = annotate_get_attrs
 
+        self._cur_node = None
+
+    def run_node(self, n):
+        try:
+            self._cur_node = n
+            return super().run_node(n)
+        finally:
+            self._cur_node = None
```
Comment (zdevito) [inline torch/fx/experimental/schema_type_annotation.py:43]: this seems a bit messy, to it is more related to the abstraction for how transforms are done and not this PR in particular.

Comment (zdevito) [reply torch/fx/experimental/schema_type_annotation.py:43]: Is there a reason this strives to preserve type for param nodes but nothing else?

**Review by zdevito — APPROVED** [review]
Seems good now, some comments unrelated to this particular change

File: torch/fx/experimental/normalize.py [hunk 3]
```diff
@@ -117,3 +120,87 @@ def _args_kwargs_to_normalized_kwargs(self, sig : inspect.Signature, args : Tupl
...
+
+        traced = NormalizeOperators(traced).transform()
+    """
+    binary_magic_method_remap : Dict[Callable[[Any, Any], Any], Callable[[Any, Any], Any]] = {
+        operator.add : torch.add,
+        operator.mul : torch.mul,
+        operator.sub : torch.sub,
+        operator.truediv : torch.div,
+        operator.floordiv: torch.floor_divide,
+        operator.mod: torch.remainder,
```
[the remark this answers is not in the document]
Comment (jamesr66a) [reply torch/fx/experimental/normalize.py]: Going in the `operator.mod` -> `torch.remainder` direction comes directly from the mapping in `python_variable_methods`: https://github.com/pytorch/pytorch/blob/fd58ececab731803eceace70b985b35f9872b53c/tools/autograd/templates/python_variable_methods.cpp#L1150 so I assume this is correct unless PyTorch has it wrong?

File: torch/fx/experimental/normalize.py (hunk 1 above)
Comment (jamesr66a) [reply torch/fx/experimental/normalize.py:148]: Would the argument for the `operator.mod -> torch.remainder` direction still apply here? This is going to get dispatched down to `torch.remainder` judging by what `python_variable_methods` does so it it consistent with PT?

File: torch/fx/experimental/schema_type_annotation.py (hunk 2 above)
Comment (jamesr66a) [reply torch/fx/experimental/schema_type_annotation.py:43]: Types were set for everything else in a previous PR, it was just missing for `placeholders`. This adds the missing annotations. Though this change is actually no longer needed now that we don't rely on types, let me just remove it
